cmd/pitr/recoverer/recoverer.go

```diff
@@ -330,6 +330,9 @@ func (r *Recoverer) setBinlogs() error {
 			return errors.Wrapf(err, "read %s gtid-set object", binlog)
 		}
 		binlogGTIDSet := string(content)
+		if r.gtidSet != strings.Split(binlogGTIDSet, ":")[0] {
```

Comment:
r.gtidSet is set by user in case of excluding some set while recovery (https://github.com/percona/percona-xtradb-cluster-operator/blob/K8SPXC-645/cmd/pitr/recoverer/recoverer.go#L98). But here you compare it with `cluster ID` from binlog file. I think you should use r.startGTID, take cluster ID from it and compare with data from binlog